Diffusion/Common/DiffusionImage.swift

```diff
@@ -0,0 +1,337 @@
+//
+//  DiffusionImage.swift
+//  Diffusion
+//
+//  Created by Dolmere on 03/07/2023.
+//
+
+import SwiftUI
+import StableDiffusion
+import CoreTransferable
+import UniformTypeIdentifiers
+#if os(macOS)
+#else
+import UIKit // Access to UIImage on iOS
+#endif
+
+/// Tracking for a `DiffusionImage` generation state.
+enum DiffusionImageState {
+    case generating
+    case waiting
+    case complete
+}
+
+/// Generic custom error to use when an image generation fails.
+enum DiffusionImageError: Error {
+    case invalidDiffusionImage
+}
+
+/// Combination of a `DiffusionImage` and its associated `DiffusionImageState`
+struct DiffusionImageWrapper {
+    var diffusionImageState: DiffusionImageState = .waiting
+    var diffusionImage: DiffusionImage? = nil
+}
+
+/// Model class  to hold a  generated image and the data used to generate it
+final class DiffusionImage: NSObject, Identifiable, NSCoding, NSSecureCoding {
+    
+    // Note: we do not capture the chosen Scheduler as it's a Swift enum and cannot conform to NSSecureCoding, which is used for Drag operations.
```

Comment:
As discussed offline, I think it's important to preserve scheduler information :)

Reply:
agreed. added in recent commit. it's been put inside the State.swift common code file.